docs/docs/tutorial/creating-a-source-plugin/part-1/index.mdx

```diff
@@ -0,0 +1,170 @@
+- A well-written `README` that explains the user in concise steps how to install, use, and configure your plugin (also see [How to write a plugin README](/contributing/docs-contributions/how-to-write-a-plugin-readme/))
+- A `src` folder with the plugin's source code
+  - A root `gatsby-node.ts` file that exports [Gatsby Node APIs](/docs/reference/config-files/gatsby-node/) from individual files
+  - Individual files that only contain the specific Node API (in [kebab case](https://en.wikipedia.org/wiki/Letter_case#Kebab_case)), e.g. `onPluginInit` becomes `on-plugin-init.ts`. This will also make unit testing your plugin easier.
+
+How you organize your `src` folder is of course up to you, we'd recommend keeping things consistent and files lowercase.
+
+<Collapsible
+  summary={<em>Wondering about a source plugin template for new projects?</em>}
+>
+
+Above we said that you can ignore the `api` folder. This is because for your real-world project you'll most probably source the data from an external API, not an API locally in your repository.
+
+The companion repository not only has a `start` branch but also a `template` branch without the `api` folder. You'll learn more about this at the end in the [What's next?](/docs/tutorial/creating-a-source-plugin/whats-next/) part. For now, go through the tutorial and at the end you'll know how to leverage the repository as a template for your new projects!
+
+</Collapsible>
+
+## Start the project
+
+Ready to see something happen in your browser? Then let's go!
+
+First, install the necessary dependencies to run everything:
+
+```shell
+yarn
+```
+
+Afterwards, open a new window in your terminal (you should have two now).
+
+In the first window, run `develop:deps` to start watching the `api` and `plugin` directories. If you change something inside `api`, the GraphQL server will be restarted, if you change something in `plugin` the TypeScript compiler will output updated files.
+
+```shell
+yarn develop:deps
+```
+
+You should see something like this in your terminal:
+
+```shell
+[develop:api   ] 14:48:03 - Starting compilation in watch mode...
+[develop:api   ]
+[develop:plugin] 14:48:03 - Starting compilation in watch mode...
+[develop:plugin]
+[develop:api   ] Server is running at http://localhost:4000/graphql
+[develop:api   ]
+[develop:api   ] 14:48:04 - Found 0 errors. Watching for file changes.
+[develop:plugin]
+[develop:plugin] 14:48:04 - Found 0 errors. Watching for file changes.
+[develop:api   ] Server is running at http://localhost:4000/graphql
+```
+
+Now, in your second terminal window, run `develop:site`. This runs `gatsby develop` for the example site:
+
+```shell
+yarn develop:site
+```
+
+While `gatsby develop` is running you should also see the `info Example plugin loaded...` log. This shows that the plugin was successfully loaded for this initial demo. Yey 🎉
```

Comment:
Small typo and a suggestion you can take or leave to avoid using sighted terms:
```suggestion
While `gatsby develop` is running you should also note the `info Example plugin loaded...` log. This verifies that the plugin was successfully loaded for this initial demo. Yay 🎉
```